### ISSUE TITLE
Add support for bip173 (segwit payment addresses).

### ISSUE THREAD

**Issue by eMosbat** [libbitcoin/libbitcoin-explorer#520]
how can I generates segwit address using bx tool?

**Comment by evoskuil** [comment]
Which type of address are you referring to? https://bitcoincore.org/en/segwit_wallet_dev

**Comment by eMosbat** [comment]
I need to generate p2wpkh addresses (if I am correct) that starts with `bc`. need a tool like bx to do it or a php script.

**Comment by eMosbat** [comment]
any good news on this? :>

**Comment by evoskuil** [comment]
I just merged a libbitcoin pull request for the base32 encoding. I know someone is working on witness_addresss as well. Once that is done it will be simple to drop in bx commands. So some progress.

**Comment by bitdaric** [comment]
I am interested too. or maybe give me a way to convert current address with a simple tool?

**Comment by anonproject** [comment]
Would you please clarify if that was merged in the 4.0 Milestone?  I am compiling the latest master now ...

**Comment by evoskuil** [comment]
It has not been implemented and the v4 milestone is not complete.

**Comment by thecodefactory** [comment]
In `master`, see `bx ec-to-witness` and `bx witness-to-key` commands.

**Comment by thecodefactory** [comment]
https://github.com/libbitcoin/libbitcoin-explorer/pull/627
https://github.com/libbitcoin/libbitcoin-system/pull/1118

**Comment by evoskuil** [comment]
Ah, I was thinking this was the issue for support of witness txs. Is this supported with the new commands?

**Comment by thecodefactory** [comment]
The BIP173 address parsing/generating is complete.  Witness tx construction isn't convenient yet though (i.e. not explicitly supported).  That is to be done, and should probably be a separate issue.

**Comment by eMosbat** [comment]
> 
> 
> In `master`, see `bx ec-to-witness` and `bx witness-to-key` commands.

could you please upload exe version?

**Comment by thecodefactory** [comment]
> > In `master`, see `bx ec-to-witness` and `bx witness-to-key` commands.
> 
> could you please upload exe version?

Sorry, I don't have a Windows environment.
